### PR TITLE
[EA] People directory posts column

### PR DESCRIPTION
**Pull request by oetherington**
Adds a posts column to the people directory. [Figma](https://www.figma.com/design/qAHRGADVOGojqKpqbGMFFu/2024-Q2?node-id=685-5490&t=kwtLu30T64XeboX8-0).

The posts are searchble, but currently we always just show the top post. This Figma says we should show the matching post if the search query matched a post name, but I'll handle that in the search highlights PR.

Screenshot 2024-07-01 at 16 20 19

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207625343155321) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
@agnesstenlund This PR isn't finished yet (I still need to make the post names searchable) but the UI is probably ready to look at.

I had some problems with the positioning of the "and x more" text for the same reasons as when we did the post excerpts in the posts tooltip. When we did that we ended up using a fade instead as it was much simpler. I tried a similar thing here but I'm not 100% convinced (maybe you have a better idea). We could also just show the entire title and not worry about shortening it to only 2 lines which would solve the issue.

Screenshot 2024-06-25 at 10 23 46

To be clear, it is _possible_ to do it as designed, but it'll be very complicated (and probably bad for performance) so I don't think I should just go ahead and spend a day or two on it unless you definitely think it's worth it.

We also have exactly the same issue in [the topics PR](#9421) which Sarah commented on. Maybe we can use the same solution there as well.

**Comment by agnesstenlund** [comment]
That definitely sounds like the right tradeoff 👍 thanks for flagging

**Comment by agnesstenlund** [comment]
Looking at this I've two new tweaks:

- [ ] Looks like the line-height is a bit tight for both post names + bio, would increase a tiny bit
Screenshot 2024-06-25 at 13 55 33

- [ ] I think the fade could look better with a "+ 12 more" instead of "and 12 more" (which would also match the topics better (my bad for making the inconsistent to begin with!)
Screenshot 2024-06-25 at 13 56 47

File: packages/lesswrong/server/repos/UsersRepo.ts [hunk 1]
```diff
@@ -172,7 +173,6 @@ class UsersRepo extends AbstractRepo<"Users"> {
 
   private getSearchDocumentQuery(): string {
     return `
-      -- UsersRepo.getSearchDocumentQuery
```
Comment (Will-Howard) [inline packages/lesswrong/server/repos/UsersRepo.ts:175]: Could you add this back in?

**Review by Will-Howard — APPROVED** [review]
LGTM

File: packages/lesswrong/server/repos/UsersRepo.ts (hunk 1 above)
Comment (oetherington) [reply packages/lesswrong/server/repos/UsersRepo.ts:175]: I'm pretty sure it's just a mistake - this function doesn't run a query, it just returns a string that gets inserted into other queries, and those other queries add their own comments.

Comment (Will-Howard) [reply packages/lesswrong/server/repos/UsersRepo.ts:175]: Ah right my bad

**Pull request merged**